ui/sv_icons.py

```diff
@@ -0,0 +1,54 @@
+import bpy
+import os
+import glob
+import bpy.utils.previews
+
+# custom icons dictionary
+_icon_collection = {}
+
+
+def custom_icon(name):
+    load_custom_icons()  # load in case they custom icons not already loaded
+
+    custom_icons = _icon_collection["main"]
+
+    default = lambda: None  # for no icon with given name will return zero
+    default.icon_id = 0
+
+    return custom_icons.get(name, default).icon_id
+
+
+def load_custom_icons():
+    if len(_icon_collection):  # return if custom icons already loaded
```

Comment:
You still have the function object, oh well, one should loop with generators not check them.
http://stackoverflow.com/questions/661603/how-do-i-know-if-a-generator-is-empty-from-the-start